test/multi-memnode-test/fam_datapath_for_memnode_regions.cpp

```diff
@@ -0,0 +1,252 @@
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+#include <fam/fam.h>
+#include <fam/fam_exception.h>
+
+#include "common/fam_test_config.h"
+
+using namespace std;
+using namespace openfam;
+#define NUM_REGIONS 10
+
+fam *my_fam;
+Fam_Options fam_opts;
+Fam_Region_Descriptor *testRegionDesc[NUM_REGIONS];
+const char *testRegionStr;
+const char *firstItem;
+Fam_Descriptor *item[NUM_REGIONS];
+
+#define REGION_PERM 0777
+#define NUM_ITERATIONS 1000
+#define REGION_SIZE 4294967296
+uint64_t gDataSize = 256;
+int numsrv;
+
+const int MAX = 26;
+const int REGION_NAME_LEN = 10;
+
+// Returns a string of random alphabets of
+// length n.
+string getRandomString() {
+    char alphabet[MAX] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i',
+                          'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r',
+                          's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
+
+    string res = "";
+    for (int i = 0; i < REGION_NAME_LEN; i++)
+        res = res + alphabet[rand() % MAX];
+
+    return res;
+}
+
+const char *getRegionName(int memServerId, int totalMemservers) {
+    size_t hashVal;
+    std::string regionName;
+    do {
+        // regionName = getRandomString()  + to_string(memServerId );
+        regionName = getRandomString();
+        hashVal = hash<string>{}(regionName) % totalMemservers;
+    } while (hashVal != (size_t)memServerId);
+    cout << "[ " << memServerId << ", " << totalMemservers
+         << " ] = " << regionName << endl;
+    return (strdup(regionName.c_str()));
+}
+
+// Test case#1 Success with minimal options.
+TEST(FamDatapath, BlockingPutGet) {
+    int j = 0;
+    char *local = strdup("MSG: ");
+    char *msg = (char *)malloc(gDataSize);
+    // allocate local memory to receive 20 elements
+    char *local2 = (char *)malloc(gDataSize);
+    uint64_t offset = 0;
+    while (j < NUM_ITERATIONS) {
+        for (int i = 0; i < numsrv; i++) {
+            // cout << "Writing to memsrver " << i << " at offset " << gDataSize
+            // * j << " for iter " << j << endl;
+            sprintf(msg, "%s %d", local, j);
+            //	    offset = (gDataSize * j) % REGION_SIZE;
+            EXPECT_NO_THROW(
+                my_fam->fam_put_blocking(msg, item[i], offset, gDataSize));
+        }
+
+        j++;
+    }
+    j = 0;
+    offset = 0;
+    while (j < NUM_ITERATIONS) {
+        for (int i = 0; i < numsrv; i++) {
+            sprintf(msg, "%s %d", local, j);
+            //	    offset = (gDataSize * j) % REGION_SIZE;
+
+            EXPECT_NO_THROW(
+                my_fam->fam_get_blocking(local2, item[i], offset, gDataSize));
+        }
+
+        j++;
+    }
+    free(msg);
+    free(local2);
+}
+
+// Test case -  Non-Blocking put test.
+TEST(FamDatapath, NonBlockingFamPut) {
+    int i, j = 0;
+    char *local = strdup("MSG: ");
+    char *msg = (char *)malloc(gDataSize);
+    uint64_t offset = 0;
+    while (j < NUM_ITERATIONS) {
+        sprintf(msg, "%s %d", local, j);
```

Comment:
In blocking put, the sprintf is inside the inner for loop. In non blocking it is outside. For consistency can both be at the same level. (preferably outside inner for loop)